overlord/configstate/config/transaction.go

```diff
@@ -343,6 +402,69 @@ func commitChange(pristine *json.RawMessage, change interface{}) *json.RawMessag
 	panic(fmt.Errorf("internal error: unexpected configuration type %T", change))
 }
 
+// mergeConfigWithVirtual takes the given configuration and merges it
+// with the virtual configuration values by calling the registered
+// virtual configuration function of the given snap. The merged config
+// is returned.
+func mergeConfigWithVirtual(instanceName string, config *json.RawMessage) (*json.RawMessage, error) {
```

Comment:
retrieving the virtual config can be relatively expensive, especially compared to memory operations, otoh it's really only needed if we are looking at a part of the config that contains virtual configs, no? so I think this should take the subkeys from the caller and call the functions only if there's an overlap. Or am I missing something?

Reply:
I added this now too.